### PR TITLE
MODEUR-14: Wire SQL to API; rework SQL

### PR DESCRIPTION
**Pull request by julianladisch**

### CODE REVIEW

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java [hunk 1]
```diff
@@ -1081,13 +1081,34 @@ static String getRanges(JsonArray ar, String startProp, String endProp) {
...
+    return getUseOverTimeJournal(pool, agreementId, start, end)
+        .map(json -> {
+          ctx.response().setStatusCode(200);
+          ctx.response().putHeader("Content-Type", "application/json");
+          ctx.response().end(json.encodePrettily());
+          return null;
+        });
+  }
+
+  Future<JsonObject> getUseOverTimeJournal(TenantPgPool pool,
```
Comment (MikeTaylor) [inline src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:1109]: It's hard for me to think that it's a good idea to give these two functions the same name. Overloading makes sense when the same-named functions are doing the same thing, but that's not the case here.

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java [hunk 2]
```diff
@@ -1081,13 +1081,34 @@ static String getRanges(JsonArray ar, String startProp, String endProp) {
...
+      String agreementId, String start, String end) {
+
     LocalDate startDate = LocalDate.parse(start).withDayOfMonth(1);
```
Comment (MikeTaylor) [inline src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:1112]: I don't know the library you're using here, but I hope its not interpreting the start- and end-date parameters as locale-appropriate formatted strings? We send ISO-format dates.

**Review by MikeTaylor — APPROVED** [review]
I gave up going through line by line, quickly realising I was out of my depth. I say go ahead and merge. It's not like this is production code.

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java (hunk 2 above)
Comment (julianladisch) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:1112]: A date without a time-zone in the ISO-8601 calendar system, such as 2007-12-03.
https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/time/LocalDate.html

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java (hunk 1 above)
Comment (julianladisch) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:1109]: They do the same thing.
This one takes the parameters out of ctx, calls the other one, and then returns the result using ctx.

File: src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java (hunk 2 above)
Comment (MikeTaylor) [reply src/main/java/org/folio/eusage/reports/api/EusageReportsApi.java:1112]: They don't do the same thing. One of them creates an HTTP response with content-type and whatnot.

**Pull request merged**